apps/api/v2/src/modules/oauth/oauth-client.repository.ts

```diff
@@ -90,24 +90,27 @@ export class OAuthClientRepository {
           },
         },
       },
+      include: {
+        authorizationTokens: {
+          where: {
+            id: tokenId,
+          },
+          include: {
+            owner: {
+              select: {
+                id: true,
+              },
+            },
+          },
+        },
+      },
     });
 
     if (!oauthClient) {
       throw new BadRequestException("Invalid Authorization Token.");
     }
 
-    const authorizationToken = await this.dbRead.prisma.platformAuthorizationToken.findUnique({
-      where: {
-        id: tokenId,
-      },
-      include: {
-        owner: {
-          select: {
-            id: true,
-          },
-        },
-      },
-    });
+    const authorizationToken = oauthClient.authorizationTokens[0];
```

Comment:
now that I think about it, it doesn't change anything since prisma will do multiple queries anyways, sorry my bad xD

I think it was better before and we just need to add a check if the authorizationToken exists and throw error if not 

sorry 🙏